src/Sarif.Viewer.VisualStudio/FileWatcher/SarifLogsMonitor.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Converters;
+using Microsoft.Sarif.Viewer.ErrorList;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.Sarif.Viewer.FileWatcher
+{
+    internal class SarifLogsMonitor : IDisposable
+    {
+        private readonly IFileSystem fileSystem;
+
+        private IDictionary<string, IFileWatcher> FileWatcherMap { get; } = new ConcurrentDictionary<string, IFileWatcher>();
+
+        internal SarifLogsMonitor(IFileSystem fs)
+        {
+            this.fileSystem = fs;
+        }
+
+        internal static SarifLogsMonitor Instance = new SarifLogsMonitor(new FileSystem());
+
+        internal void StartWatch(string logFilePath)
+        {
+            if (!fileSystem.FileExists(logFilePath))
+            {
+                return;
+            }
+
+            string logPath = logFilePath.ToLower();
+            if (!FileWatcherMap.ContainsKey(logPath))
+            {
+                var watcher = new FileWatcher(logFilePath);
+                watcher.SarifLogFileChanged += this.Watcher_SarifLogFileChanged;
+                watcher.SarifLogFileRenamed += this.Watcher_SarifLogFileRenamed;
+                FileWatcherMap.Add(logPath, watcher);
+                watcher.Start();
+            }
+        }
+
+        internal void Clear()
+        {
+            if (FileWatcherMap.Any())
+            {
+                FileWatcherMap.Values.ToList().ForEach(w => w.Stop());
```

Comment:
>FileWatcherMap.Values.ToList().ForEach(w => w.Stop()); [](start = 16, length = 54)

You could replace to something like this:
foreach (IFileWatcher item in FileWatcherMap.Values)
                {
                    (item as IDisposable)?.Dispose();
                }

Reply:
Fixed